### PR TITLE
add flatten-dict

### PR DESCRIPTION
**Pull request by bgruening**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/flatten-dict/LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
+MIT License
```
Comment (synapticarbors) [inline recipes/flatten-dict/LICENSE:1]: Please remove license file. It's already in the tarball

File: recipes/flatten-dict/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 0ccc43f15c7c84c5ef387ad19254f6769a32d170313a1bcbf4ce582089313d7e
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
```
Comment (synapticarbors) [inline recipes/flatten-dict/meta.yaml:18]: It looks like the `host` section requires `poetry`

File: recipes/flatten-dict/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - pip
+    - python
+    - poetry
+  run:
+    - pathlib2 >=2.3,<3.0
```
Comment (synapticarbors) [inline recipes/flatten-dict/meta.yaml:23]: it looks like you need `setuptools` under `run` due to importing `pkg_resources`

**Comment by bgruening** [comment]
@synapticarbors thanks for the review!

**Pull request merged**